lib/Dialect/TTIR/Transforms/Reshape.cpp

```diff
@@ -0,0 +1,53 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "ttmlir/Dialect/TT/IR/TT.h"
+#include "ttmlir/Dialect/TTIR/Transforms/Passes.h"
+
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+#include <mlir/Transforms/GreedyPatternRewriteDriver.h>
+
+namespace mlir::tt::ttir {
+#define GEN_PASS_DEF_TTIRRESHAPEFOLD
+#include "ttmlir/Dialect/TTIR/Transforms/Passes.h.inc"
+
+//===----------------------------------------------------------------------===//
+// Constant as fill pass
+//===----------------------------------------------------------------------===//
+
+class TTIRReshapeFoldingRewriter : public OpRewritePattern<ReshapeOp> {
+public:
+  using OpRewritePattern<ReshapeOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(ReshapeOp op,
+                                PatternRewriter &rewriter) const final {
+    if (op.getType() == op->getOperand(0).getType()) {
+      rewriter.replaceOp(op, op->getOperand(0));
+      return success();
+    }
+    return failure();
+  }
+};
+
+class TTIRReshapeFold : public impl::TTIRReshapeFoldBase<TTIRReshapeFold> {
```

Comment:
@azecevicTT This seems like a good candidate for a canonicalization...

Reply:
Yeah, these kind of things should be covered with canonicalization in the future. TableGen seems like a more suitable option for these kind of simple patterns, because it supports a proper pattern matching.

Reply:
@azecevicTT @sdjordjevicTT Something to take a look at for future patterns: https://mlir.llvm.org/docs/PDLL/. MLIR now supports defining patterns using PDLL in addition to tablegen. Somewhere in that page they give some reasoning as to why they made PDLL instead of just only having tablegen.